### ISSUE TITLE
Running "pacaur -Syu" does not detect out-of-date packages

### ISSUE THREAD

**Issue by lasers** [rmarquis/pacaur#350]
I noticed this when I started trying couple of yaourt commands after a long time of using pacaur as my aur helper. Before this log, I ran "pacaur -Syu" between yaourt commands to see what pacaur would list and it does list something -- One out-of-date package ( gtk-theme-arc-git ). Yaourt did detect that gtk-theme-arc have been updated (and wasn't on the list anymore). Is this a proper behaviour or am I missing something simple?
```
local/pacaur 4.2.27-1
local/yaourt 1.6-1
```
```
chris ~ $ pacaur -Syyua
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 antergos is up to date
:: Starting AUR upgrade...
 there is nothing to do
```
```
chris ~ $ yaourt -Syyua
:: Synchronizing package databases...
 core                              121.7 KiB   795K/s 00:00 [######################] 100%
 extra                            1741.9 KiB  3.59M/s 00:00 [######################] 100%
 community                           2.7 MiB  4.13M/s 00:01 [######################] 100%
 multilib                          125.9 KiB  4.10M/s 00:00 [######################] 100%
 antergos                          415.6 KiB   474K/s 00:01 [######################] 100%
 Foreign packages: - 150 / 150

==> Package upgrade only (new release):
aur/gnome-colors-icon-theme        5.5.1-2               2 -> 3
aur/gnome-colors-icon-theme-extras 5.5.1-2               2 -> 3
aur/gtk2-theme-dust                0.5.1-1               1 -> 2
aur/gtk2-theme-dust-extras         0.5.1-1               1 -> 2
aur/python2-ujson                  1.33-1                1 -> 2
aur/shimmer-wallpapers             1.0-2                 2 -> 4
aur/solaar                         0.9.2-1               1 -> 2
aur/xcb-util-cursor-git            1:0.1.2.r0.g8eb844d-1 1 -> 2
aur/zathura-cb                     0.1.4-2               2 -> 3

==> Software upgrade (new version) :
aur/antigen-git                    r335.30098c1-1        -> r358.7860ce7-1
aur/flattr-icon-theme-git          552.e7670d0-1         -> 560.222f37e-1
aur/i3status-git                   2.9.r22.g3545e03-1    -> 2.9.r37.gadaa832-1
aur/j4-dmenu-desktop-git           2.12.r4.g666dd0b-1    -> 2.12.r9.gebe007a-1
aur/liblxqt-git                    0.9.0.8.ga844118-1    -> 0.9.0.15.gae45471-1
aur/libqtxdg-git                   1.2.0.12.gd5bc16b-1   -> 1.2.0.13.gab0f65c-1
aur/lolcat                         42.0.99-3             -> 42.1.0-3
aur/paccache                       0.1.14-1              -> 0.1.15-1
aur/pcmanfm-qt-git                 0.9.0.9.g2a4ec48-1    -> 0.9.0.14.g643dcf6-1
aur/python-click                   4.0-2                 -> 4.1-1
aur/python2-trollius               1.0.4-2               -> 2.0-1
aur/rofi-git                       r764.f14efce-1        -> 0.15.5.r38.g9b83c61-1
aur/ruby-rspec                     3.1.0-1               -> 3.3.0-1
aur/ruby-rspec-core                3.1.7-1               -> 3.3.0-1
aur/ruby-rspec-expectations        3.1.2-1               -> 3.3.0-1
aur/ruby-rspec-mocks               3.1.3-1               -> 3.3.0-1
aur/ruby-rspec-support             3.1.2-1               -> 3.3.0-1
aur/visual-studio-code             0.3.0-1               -> 0.5.0-3
aur/xcape-git                      1.1.9.gf3802fc-1      -> 1:1.1.r9.gf3802fc-2

==> Continue upgrade ? [Y/n]
==> [V]iew package detail   [M]anually select packages
==> --------------------------------------------------
==> 
```

**Comment by rmarquis** [comment]
The above output only tell me you've done all AUR update before running pacaur. In other words, I have no idea what you're writing about. I don't even see any ```gtk-theme-arc``` in the yaourt output above.

Please give more details.

**Comment by lasers** [comment]
Hi rmarquis. I will break down what I did.

Ran pacaur commands for nearly everyday since March 3 (installation date).
Ran yaourt command for first time ever. 20+~ out-of-date packages. Cancelled.
Ran pacaur command. 1 out-of-date package. Updated gtk-theme-arc-git.
Ran yaourt command one more time. 19+~ out-of-date packages. Scrolled up the scrollback.
Compared 2nd yaourt command with 1st yaourt command. No gtk-theme-arc-git.

How do I update the missing 19+~ out-of-date packages that yaourt picked up?

I filed "Pacuar does not detect out-of-date packages." and wanted to know if I'm missing something simple...  because that's the result of running "yaourt -Syyua" exactly right after running "pacaur -Syyua".

TL:DR: yaourt detects more out-of-packages than pacaur.

**Comment by rmarquis** [comment]
All right, now I understand what you mean.

Do you happen to use AUR4 with yaourt? Pacaur defaults to AUR3. As far as I see, the output above looks like yaourt tries to update obsolete package in AUR3 to the newer version found in AUR4.

If you want to use AUR4 with pacaur, do the following:
- install latest ```cower-git```
- add an alias in your .bashrc: 

```
alias pacaur4='pacaur --domain aur4.archlinux.org'
```

* source with ```. ~/.bashrc```
* then try to update again with ```pacaur4 -Syu```.

Note that AUR4 will become the default on August 8, so this alias will stop working at that time.

**Comment by lasers** [comment]
I had cower-git installed. I have to install pacaur-git too. Both looks same at a glance (versions). There are many same package name (one and one without -git) with either same/different maintainers... even with orphaned packages. Heh. I overlooked the *-git part on Archlinux Pacaur Wiki. Sorry.
```
chris ~ $ pacaur -Ss pacaur
aur/pacaur 4.2.27-1 (314)
    A fast workflow AUR helper using cower as backend
aur/pacaur-git 4.2.27-1 (17) [installed: 4.2.27.21.geb10499-1]
    A fast workflow AUR helper using cower as backend
```
Now using pacaur-git and cower-git.... Installing missing packages!
```
chris ~ $ pacaur --domain aur4.archlinux.org -Syu 
[sudo] password for chris: 
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 antergos is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Starting AUR upgrade...
:: archey is not present in AUR -- skipping
:: base16-shell-git is not present in AUR -- skipping
:: chromo-erectus-icon-themes is not present in AUR -- skipping
:: corebird-git is not present in AUR -- skipping
:: gcolor3 is not present in AUR -- skipping
:: gnome-color-chooser is not present in AUR -- skipping
:: gohufont is not present in AUR -- skipping
:: gtk-theme-numix-osx is not present in AUR -- skipping
:: gtk-theme-shiki-colors-flat is not present in AUR -- skipping
:: htop-vi is not present in AUR -- skipping
:: humanity-icon-theme is not present in AUR -- skipping
:: irssi-scripts-git is not present in AUR -- skipping
:: j4-make-config-git is not present in AUR -- skipping
:: node-webkit is not present in AUR -- skipping
:: numix-themes-darkblue is not present in AUR -- skipping
:: numix-themes-tomorrow is not present in AUR -- skipping
:: numix-themes-yellow is not present in AUR -- skipping
:: pipes-git is not present in AUR -- skipping
:: pkgtools is not present in AUR -- skipping
:: python-i3-git is not present in AUR -- skipping
:: python-neovim-git is not present in AUR -- skipping
:: python-virtkey is not present in AUR -- skipping
:: python2-wsgiref is not present in AUR -- skipping
:: quickswitch-i3-git is not present in AUR -- skipping
:: retroarch-git is not present in AUR -- skipping
:: retroarch-phoenix-qt-git is not present in AUR -- skipping
:: rmlint-git is not present in AUR -- skipping
:: sxhkd-git is not present in AUR -- skipping
:: termsyn is not present in AUR -- skipping
:: tightvnc is not present in AUR -- skipping
:: trash-cli is not present in AUR -- skipping
:: tv-maxe is not present in AUR -- skipping
:: twd is not present in AUR -- skipping
:: ultra-flat-gtk is not present in AUR -- skipping
:: urxvt-autocomplete-all-the-things-git is not present in AUR -- skipping
:: urxvt-font-size-git is not present in AUR -- skipping
:: vlc-skins is not present in AUR -- skipping
:: weatherboy is not present in AUR -- skipping
:: wine-silverlight is not present in AUR -- skipping
:: winusb is not present in AUR -- skipping
:: xcursor-human is not present in AUR -- skipping
:: zgen-git is not present in AUR -- skipping
:: resolving dependencies...
:: looking for inter-conflicts...
:: antigen-git latest revision -- fetching
:: flattr-icon-theme-git latest revision -- fetching
:: i3status-git latest revision -- fetching
:: j4-dmenu-desktop-git latest revision -- fetching
:: liblxqt-git latest revision -- fetching
:: libqtxdg-git latest revision -- fetching
:: pcmanfm-qt-git latest revision -- fetching
:: rofi-git latest revision -- fetching
:: xcape-git latest revision -- fetching
:: xcb-util-cursor-git latest revision -- fetching

AUR Packages  (28):

Name                                Old Version            New Version          

aur/antigen-git                     r335.30098c1-1         latest                              
aur/flattr-icon-theme-git           552.e7670d0-1          latest                              
aur/gnome-colors-icon-theme         5.5.1-2                5.5.1-3                             
aur/gnome-colors-icon-theme-extras  5.5.1-2                5.5.1-3                             
aur/gtk2-theme-dust                 0.5.1-1                0.5.1-2                             
aur/gtk2-theme-dust-extras          0.5.1-1                0.5.1-2                             
aur/i3status-git                    2.9.r22.g3545e03-1     latest                              
aur/j4-dmenu-desktop-git            2.12.r4.g666dd0b-1     latest                              
aur/liblxqt-git                     0.9.0.8.ga844118-1     latest                              
aur/libqtxdg-git                    1.2.0.12.gd5bc16b-1    latest                              
aur/lolcat                          42.0.99-3              42.1.0-3                            
aur/paccache                        0.1.14-1               0.1.15-1                            
aur/pcmanfm-qt-git                  0.9.0.9.g2a4ec48-1     latest                              
aur/python-click                    4.0-2                  4.1-1                               
aur/python2-trollius                1.0.4-2                2.0-1                               
aur/python2-ujson                   1.33-1                 1.33-2                              
aur/rofi-git                        r764.f14efce-1         latest                              
aur/ruby-rspec                      3.1.0-1                3.3.0-1                             
aur/ruby-rspec-core                 3.1.7-1                3.3.0-1                             
aur/ruby-rspec-expectations         3.1.2-1                3.3.0-1                             
aur/ruby-rspec-mocks                3.1.3-1                3.3.0-1                             
aur/ruby-rspec-support              3.1.2-1                3.3.0-1                             
aur/shimmer-wallpapers              1.0-2                  1.0-4                               
aur/solaar                          0.9.2-1                0.9.2-2                             
aur/visual-studio-code              0.3.0-1                0.5.0-3                             
aur/xcape-git                       1.1.9.gf3802fc-1       latest                              
aur/xcb-util-cursor-git             1:0.1.2.r0.g8eb844d-1  latest                              
aur/zathura-cb                      0.1.4-2                0.1.4-3                             

:: Proceed with installation? [Y/n] 
```
Yes, I forget that I modified /etc/yaourtrc to troubleshoot somebody with AUR4 issue on yaourt. Everything is working perfectly now. Many thanks for pacaur. It is one of my favourite tools.

**Comment by rmarquis** [comment]
No problem, you're very welcome.
